utils-for-endpoints/storage/storage.go

```diff
@@ -0,0 +1,190 @@
+package storage
+
+import (
+	"errors"
+	"github.com/button-tech/utils-node-tool/shared/db"
+	"github.com/button-tech/utils-node-tool/shared/db/schema"
+	"github.com/imroc/req"
+	"github.com/onrik/ethrpc"
+	"log"
+	"math/rand"
+	"os"
+	"runtime"
+	"sync"
+	"time"
+)
+
+type StoredEndpoints struct {
+	sync.RWMutex
+	Entry schema.EndpointsData
+}
+
+func (s *StoredEndpoints) Set(entry schema.EndpointsData) {
+	s.Lock()
+	s.Entry = entry
+	s.Unlock()
+}
+
+func (s *StoredEndpoints) Get() *schema.EndpointsData {
+	s.RLock()
+	defer s.RUnlock()
+	return &s.Entry
+}
+
+type FastestEndpoint struct {
+	sync.RWMutex
+	Address string
+}
+
+func (f *FastestEndpoint) Set(addr string) {
+	f.Lock()
+	f.Address = addr
+	f.Unlock()
+}
+
+func (f *FastestEndpoint) Get() string {
+	f.RLock()
+	defer f.RUnlock()
+	return f.Address
+}
+
+type GetFastestEndpoint func(string) string
+
+var (
```

Comment:
Instead 2 global var make 2 constructors and 2 interfaces like this:
```
type EndpointsStorage interface {
     //Your methods here
    //Get()
    //Set()
}

type FastestEndpointStorage interface {
     //Your methods here
    //Get()
    //Set()
}
```

```
func NewEndpointsStoreConstructor (EndpointsStorage) {
	return {
		mu: new(sync.Mutex),
		store: new(StoredEndpoints),
	}
}

func NewFastestEndpointStoreConstructor (FastestEndpointStorage) {
	return {
		mu: new(sync.Mutex),
		store: new(StoredEndpoints),
	}
}
```

So, in main.go you should do like this:

```
fastestStore := NewFastestEndpointStoreConstructor()
endPointsStore := NewEndpointsStoreConstructor()

// throw each of storage in every paths in your program, but it maybe need some refactors your project-model
//For example, you should throw it in your handlers:

// handlers.go
type controller struct {
  storage FastestEndpointStorage
}

func(c controller) SomeFunc(c *gin.Context) {
  // Get Request

 // You should answer with data from storage :
   fromStore := c.storage.Get()
   c.JSON(200, gin.H{"result": &fromStore}
}
```

Reply:
Thank you, I'll think about it